ftlid/__init__.py

```diff
@@ -5,22 +5,30 @@
 DIRNAME = os.path.dirname(os.path.abspath(__file__))
 
 
-def identify_language(text, model_path=None, with_prob=False, **kwargs):
+def identify_language(text, model=None, model_path=None, with_prob=False,
+                      **kwargs):
     """
     Identifies language the provided text is written in using a pre-trained
-    fastText model. Except for the arguments listed below, anky keyword
-    arguments are fassed further to fastText's predict function.
+    fastText model. Except for the arguments listed below, any keyword
+    arguments are passed further to fastText's predict function.
 
     Args:
         - text: The text whose language is to be identified.
+        - model (optional): Provides an instance of a fastText model used for
+          prediction. Maximum one of the model and model_path arguments can be
+          provided.
         - model_path (optional): Provides path to the fastText model used for
           prediction. Defaults to lid.176.ftz which is included in the package.
+          Maximum one of the model and model_path arguments can be provided.
         - with_prob (optional): If True, the function returns both predicted
           languages and their probabilities. Defaults to False.
     """
-    if model_path is None:
-        model_path = DIRNAME + '/lid.176.ftz'
-    model = fasttext.load_model(model_path)
+    if model is None:
+        model = load_model(model_path)
```

Comment:
@swecooo I could see how in here the model could be loaded globally -- that is when `model is None`, the model would not get loaded from the disk but from say a global `model` variable which would get saved in the scope of this module.

Do you think something like that may be useful?

Reply:
In my opinion it would be very useful to avoid loading the model on every request by default. So I would suggest following steps:

1. If `model` is provided, use model variable.
1. If `model` is not provided but `model_path` is, load model from `model_path`.
1. If neither `model` nor `model_path` is provided, load default model from global variable.

Do you agree on these steps?

Reply:
Sure @swecooo, makes sense to me!